### PR TITLE
fixes #11990 improve virtual interfaces discovery in fact parser

### PR DESCRIPTION
**Pull request by roidelapluie**

### CODE REVIEW

**Comment by domcleal** [comment]
Please note that there's a test failure, you can see the details on http://ci.theforeman.org/job/test_develop_pr_core/6421/.  It'd be good if you can also add a test specifically to demonstrate this behaviour - thanks!

File: test/unit/fact_parser_test.rb [hunk 1]
```diff
@@ -79,6 +79,26 @@ class FactParserTest < ActiveSupport::TestCase
     end
   end
 
+  test "#find_physical_interface finds physical intervaces" do
+    parser.stub(:get_interfaces, ['eth0', 'eth0.0', 'br0', 'bond0']) do
+      result = parser.send(:find_physical_interface)
```
Comment (domcleal) [inline test/unit/fact_parser_test.rb]: I think you'll either need to pass the array of interface names (`['eth0'...]`) into find_physical_interface by doing `result = parser.send(:find_physical_interface, ['eth0'....])` or to call `parser.interfaces` like the test above.

Comment (roidelapluie) [inline test/unit/fact_parser_test.rb]: I am on it :-)

Comment (roidelapluie) [inline test/unit/fact_parser_test.rb]: better tests now

**Comment by roidelapluie** [comment]
@domcleal this is for a major version. Release notes should include:

Fixed a bug about the detection of virtual interfaces. It impacted the autodetection of the primary interface, which in the past prioritized bonds over other virtual interfaces in the autodection order.

**Comment by dLobatog** [comment]
@roidelapluie :+1:  It looks nearly ready to me. Could you refactor the tests a bit? It seems like you're constantly defining stubs and expectations you could easily do in a `setup` block.  If you use a context you can do that and restrict the expectations just for your tests.

Thanks again, great catch!

**Comment by roidelapluie** [comment]
@eLobato I could only do that for 2 tests

**Comment by dLobatog** [comment]
Merged as 01229efb92c8da85b0b918f1b7f3fd63da54fae8, thanks @roidelapluie!

**Pull request closed without merging**